src/NodeAnalyzer/CallLikeExpectsThisBindedClosureArgsAnalyzer.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\NodeAnalyzer;
+
+use PhpParser\Node\Arg;
+use PhpParser\Node\Expr\CallLike;
+use PhpParser\Node\Expr\Closure;
+use PHPStan\Reflection\ParameterReflectionWithPhpDocs;
+use Rector\NodeTypeResolver\PHPStan\ParametersAcceptorSelectorVariantsWrapper;
+use Rector\PHPStan\ScopeFetcher;
+use Rector\Reflection\ReflectionResolver;
+
+final class CallLikeExpectsThisBindedClosureArgsAnalyzer
+{
+    public function __construct(
+        private readonly ReflectionResolver $reflectionResolver
+    ) {
+    }
+
+    /**
+     * @param CallLike $callLike
+     * @return Arg[]
+     * @throws \Rector\Exception\ShouldNotHappenException
+     */
+    public function getArgsUsingThisBindedClosure(CallLike $callLike): array
+    {
+        $args = [];
+        $reflection = $this->reflectionResolver->resolveFunctionLikeReflectionFromCall($callLike);
+
+        if ($callLike->isFirstClassCallable()) {
+            return [];
+        }
+
+        if ($reflection === null) {
+            return [];
+        }
+
+        $scope = ScopeFetcher::fetch($callLike);
+
+        $parametersAcceptor = ParametersAcceptorSelectorVariantsWrapper::select($reflection, $callLike, $scope);
+        $parameters = $parametersAcceptor->getParameters();
+
+        foreach ($callLike->getArgs() as $index => $arg) {
+            if (! $arg->value instanceof Closure) {
+                continue;
+            }
+
+            if ($arg->name?->name !== null) {
+                foreach ($parameters as $parameter) {
+                    /** @phpstan-ignore method.notFound */
```

Comment:
I prefer to not inline ignore as possible, use `@var` on line 43 if needed where parameters is defined.

Reply:
in that case, in loop, you can check conditional:

```php
if (! $parameter instanceof \PHPStan\Reflection\ExtendedParameterReflection) {
    continue;
}
```

Reply:
Thanks, that appears to be working.